dig.go

```diff
@@ -107,12 +108,12 @@ func (c *Container) Provide(constructor interface{}, opts ...ProvideOption) erro
 	return nil
 }
 
-// Invoke runs a function, supplying its arguments from the Container. If the
-// function's last return value is an error, that error is propagated to the
-// caller. All other returned values (if any) are ignored.
+// Invoke runs the given function immediately, supplying its arguments from
```

Comment:
Specify "immediately".

- [ ] Before it returns?
- [ ] Order of instantiation of dependencies?

Describe how its arguments are dependencies and causes Dig to retain an instance of all its transitive dependencies. Consider well known terms like "roots" of the DAG.